File: backend/src/main/java/com/galinho/backend/dto/ServicoCreate.java
package com.galinho.backend.dto;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import com.galinho.backend.model.Financeiro.PagamentoServico;
import com.galinho.backend.utils.TipoStatus;

import jakarta.validation.constraints.Null;


public record ServicoCreate(
    @Null Long id,
    TipoStatus status,
    LocalDateTime dataInicio,
    LocalDateTime dataFim,
    LocalDateTime dataPrevisao,
    String descricao,
    BigDecimal orcamento,
    PagamentoServico pagamento,
    String veiculoPlaca
    //Long veiculoId
    //Tarefa conjuntoTarefas
    //ProdutoServico produtosUsados
    //Sem mecanicos
) {
    
}

File: frontend/src/interfaces/ServicoCreate.ts
import type { Veiculo } from "./Veiculo";
import type { Mecanico } from "./Mecanico";
import type { PagamentoServico } from "./PagamentoServico";
import type { ProdutoServico } from "./ProdutoServico";
import type { Tarefa } from "./Tarefa";

export interface ServicoCreate {
    // id?: number;
    status: string;
    dataInicio: string | null;
    // dataFim: Date;
    dataPrevisao: string | null;
    descricao: string;
    orcamento: number;
    // pagamento: PagamentoServico;
    // mecanicos: Mecanico[];
    veiculoPlaca: string;//veiculoId: number;
    // produtosUsados: ProdutoServico[];
    // conjuntoTarefas: Tarefa;
}

